contracts/cw1-subkeys/src/multitest/admin_vote_contracts.rs

```diff
@@ -0,0 +1,295 @@
+pub const QUORUM: Item<Decimal> = Item::new("quorum");
+pub const PROPOSED_ADMIN: Item<Addr> = Item::new("proposed_admin");
+pub const VOTE_OWNER: Item<Addr> = Item::new("vote_owner");
+pub const START_TIME: Item<Timestamp> = Item::new("start_time");
+pub const REQUIRED_VOTES: Item<Decimal> = Item::new("required_approvals");
+pub const PENDING_VOTES: Map<Addr, Addr> = Map::new("pending_votes");
+
+pub const VOTE_INSTANTIATE_ID: u64 = 1;
+
+pub struct AdminContract {}
+
+pub struct VoteContract {}
+
+#[derive(Serialize, Deserialize, Eq, PartialEq, Debug, Clone, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct AdminsListResp {
+    pub admins: Vec<Addr>,
+}
+
+#[derive(Serialize, Deserialize, Eq, PartialEq, Debug, Clone, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct ProposeAdminResp {
+    pub vote_addr: Addr,
+}
+
+#[contract(module=admin, error=StdError)]
+impl AdminContract {
+    #[msg(instantiate)]
+    pub fn instantiate(
+        &self,
+        ctx: (DepsMut, Env, MessageInfo),
+        admins: Vec<String>,
+        vote_code_id: u64,
+        quorum: Decimal,
+    ) -> StdResult<Response> {
+        let (deps, env, _) = ctx;
+        for addr in admins.into_iter() {
+            ADMINS.save(
+                deps.storage,
+                deps.api.addr_validate(&addr)?,
+                &env.block.time,
+            )?;
+        }
+        VOTE_CODE_ID.save(deps.storage, &vote_code_id)?;
+        QUORUM.save(deps.storage, &quorum)?;
+        Ok(Response::new())
+    }
+
+    #[msg(exec)]
+    pub fn propose_admin(
+        &self,
+        ctx: (DepsMut, Env, MessageInfo),
+        admin_code_id: u64,
+        addr: String,
+    ) -> StdResult<Response> {
+        let (deps, _env, info) = ctx;
+        let msg = vote::InstantiateMsg {
+            quorum: QUORUM.load(deps.storage)?,
+            proposed_admin: addr,
+            admin_code_id,
+        };
+
+        let msg = WasmMsg::Instantiate {
+            admin: None,
+            code_id: VOTE_CODE_ID.load(deps.storage)?,
+            msg: to_binary(&msg)?,
+            funds: vec![],
+            label: format!("admin-{}", info.sender),
+        };
+
+        let resp = Response::new()
+            .add_submessage(SubMsg::reply_on_success(msg, VOTE_INSTANTIATE_ID))
+            .add_attribute("action", "propose_admin")
+            .add_attribute("sender", info.sender);
+
+        Ok(resp)
+    }
+
+    #[msg(query)]
+    pub fn admins_list(&self, ctx: (Deps, Env)) -> StdResult<AdminsListResp> {
+        let (deps, _) = ctx;
+        let admins: Vec<Addr> = ADMINS
+            .range(deps.storage, None, None, cosmwasm_std::Order::Ascending)
+            .filter_map(|admin| admin.ok())
+            .map(|(addr, _)| addr)
+            .collect();
+        let resp = AdminsListResp { admins };
+        Ok(resp)
+    }
+
+    #[msg(reply)]
+    pub fn vote_instantiated(&self, ctx: (DepsMut, Env), msg: SubMsgResult) -> StdResult<Response> {
```

Comment:
I assume you wanted to test the reply, and that is why you put this contract. But contracts are more like examples, not tests. We want to test in `sylvia/tests` where you put a fake contract testing particular utilities. This contract is ok, but extract it as a separate contract, and test it there (it can easily depend on cw1).

Also - because you don't do any smart dispatching, this function should be called `reply` as you handle any replies here, not only `vote_instantiated` (which you expect, but technically you need to verify). Also - I don't see reply id verification. I know that you expect only one reply here, but it is bad habit to assume it, as sometimes it might change in the future.